### PR TITLE
Add updated at and created at to index

### LINKED ISSUE #162: Add `updatedAt` and `createdAt` to `Index`
When manipulating indexes (get/create/update), the MeiliSearch server returns the following information related to the index:
- `uid`: the uid of the index
- `primaryKey`: the primary key of the index
- `updateAt`: when the index has been updated
- `createdAt`: when the index has been created

This SDK, when manipulating indexes (get/create/update) returns an instance of `Index` which already contains the `primaryKey` and `uid` members.

It means we can currently get the `primaryKey` and the `uid`, but not `updatedAt` and `createdAt`.

TODO:
- [ ] Add `updatedAt` and `createdAt` member in the `Index` class/struct (depending on the language).
- [ ] Update the variable in the code where `primaryKey` is already updated. Ex: when getting the index with `getIndex()`
- [ ] Add tests

⚠️ This issue is generated, it means `primaryKey`, `updatedAt`, and `createdAt` might be named differently in this package (ex: `primary_key` instead). Keep the already existing way of naming in this package to stay idiomatic with the language and the repository.
Also, if you are a maintainer, feel free to add any clarification and instruction about this issue related to the specific languages of the repo.

Related to https://github.com/meilisearch/integration-guides/issues/121

Sorry if this is already partially/completely implemented, feel free to let me know about the state of this issue in the repo.

### PR DESCRIPTION
**Pull request by sander1095**
This PR attempts to add the `UpdatedAt` and `CreatedAt` properties to the dotnet library.

Resolves #162

### CODE REVIEW

**Comment by sander1095** [comment]
You can see the unit tests fail.

Also, when I run the project locally when I get the following error when trying to create a new index:

```
Json deserialize error: unknown field `updatedAt`, expected `uid` or `primaryKey` at line 1 column 59,
```

I am kinda confused and stuck with this, I thought the server would already be updated and ready for these properties?

Any guidance would be appreciated. 

/cc @curquiza  could you be of any assistance? I will also post this question in the Slack for help of other (community) members :)

**Comment by sander1095** [comment]
Carolina told me I could also ping

/cc @alallema

**Comment by alallema** [comment]
> You can see the unit tests fail.
> 
> Also, when I run the project locally when I get the following error when trying to create a new index:
> 
> ```
> Json deserialize error: unknown field `updatedAt`, expected `uid` or `primaryKey` at line 1 column 59,
> ```
> 
> I am kinda confused and stuck with this, I thought the server would already be updated and ready for these properties?
> 
> Any guidance would be appreciated.
> 
> /cc @curquiza could you be of any assistance? I will also post this question in the Slack for help of other (community) members :)

Hi @sander1095,
I think the confusion came from these two fields `updateAt` and `createdAt` are generated by Meilisearch API, so you can `GET` them but the `Json deserialize error` means that you try to `POST` them to Meilisearch when you can't.
I let you check the [documentation](https://docs.meilisearch.com/reference/api/indexes.html#get-one-index) to see the difference between `get` and `create`.
The `updatedAt` and `createdAt` should be in the `Index` class but maybe not in the `Index` function depending on how the SDK is made.

**Comment by sander1095** [comment]
Thanks, that might just give me enough info to look further into this! Expect more updates this evening or perhaps this weekend

**Comment by sander1095** [comment]
Hello @alallema  ! I have been able to resolve the issue, thanks to your quick reply!

I am curious about your review! I know Carolina said you people are very busy, but I can't resist asking 🙈 !

Thanks for the opportunity to create this PR! It was fun! :)

**Review by curquiza — CHANGES_REQUESTED** [review]
Can you also updat `UpdateIndex()` in the Index class as you updated `FetchInfo()`? :) Thanks

**Comment by curquiza** [comment]
bors try

**Comment by curquiza** [comment]
looks like the tests do not pass anymore, small changes might be needed :)

**Comment by sander1095** [comment]
@curquiza I have made the requested changes 🥳

**Comment by sander1095** [comment]
I tried doing `bors try` but I don't have permission :)

**Comment by curquiza** [comment]
bors try

File: src/Meilisearch/MeilisearchClient.cs [hunk 1]
```diff
@@ -74,7 +75,10 @@ public Index Index(string uid)
         public async Task<Index> CreateIndex(string uid, string primaryKey = default)
         {
             Index index = new Index(uid, primaryKey);
-            var response = await this.http.PostAsJsonAsync("/indexes", index);
+
+            JsonSerializerOptions options = new JsonSerializerOptions { IgnoreNullValues = true, PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+
+            var response = await this.http.PostAsJsonAsync("/indexes", index, options);
```
Comment (curquiza) [inline src/Meilisearch/MeilisearchClient.cs]: I think you should use `PostJsonCustomAsync`.
https://github.com/meilisearch/meilisearch-dotnet/blob/ad5ea3013528ca0b6a0ea59caba13af6de18b660/src/Meilisearch/Extensions/HttpExtensions.cs#L30-L44

Comment (sander1095) [reply src/Meilisearch/MeilisearchClient.cs]: Done.  Please check again.

Also note that I moved the `PropertyNamingPolicy = JsonNamingPolicy.CamelCase,` to inside the if block. I thought that if the user specifies different json serializer settings, we might not want to override their choices?

How do you see this?

**Review by curquiza — CHANGES_REQUESTED** [review]
Please don't modify `PrepareJsonPayload`, we don't want the `IgnoreNullValues` option be applied to every call :)

Also, it seems to have some git conflict you need to solve before I can review again!

thank you again for your involvement!

File: src/Meilisearch/MeilisearchClient.cs [hunk 2]
```diff
@@ -75,7 +75,10 @@ public Index Index(string uid)
         public async Task<Index> CreateIndex(string uid, string primaryKey = default)
         {
             Index index = new Index(uid, primaryKey);
-            var response = await this.http.PostAsJsonAsync("/indexes", index);
+
+            var options = new JsonSerializerOptions { IgnoreNullValues = true };
+
```
Comment (curquiza) [inline src/Meilisearch/MeilisearchClient.cs]:
```suggestion
            var options = new JsonSerializerOptions { IgnoreNullValues = true };
```

**Review by curquiza — APPROVED** [review]
Thank you so much @sander1095 

bors merge

I think I already sent the form link, but just in case:
If you are participating in [Hacktoberfest](https://blog.meilisearch.com/contribute-hacktoberfest-2021/), and you would like to receive a small gift from MeiliSearch too, please complete [this form](https://meilisearch.typeform.com/to/rPexR5gq).

**Comment by curquiza** [comment]
bors merge-

**Comment by curquiza** [comment]
bors merge

**Comment by curquiza** [comment]
Since bors seems to be broken I merge manually.

**Pull request merged**